core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,6 +64,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // TODO(davies): configurable
+  if (rdds.length > 10) {
+    // prepare the partitions in parallel (will be cached)
+    val parArray = rdds.toParArray
+    parArray.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(10))
+    parArray.foreach(_.partitions)
+  }
+
```

Comment:
Should we move it in `getPartitions`? Also, it will be great if we do not rely on any implicit semantic of parallel collection. I mean that we call `partitions` in parallel and get all results in a single collection explicitly.